admin/installer/cloudformation.py

```diff
@@ -70,19 +70,29 @@ def _sibling_lines(filename):
 # Keys corresponding to CloudFormation user Inputs.
 access_key_id_param = template.add_parameter(Parameter(
     "AmazonAccessKeyID",
-    Description="Your Amazon AWS access key ID",
+    Description="Your Amazon AWS access key ID (mandatory)",
     Type="String",
+    NoEcho=True,
+    AllowedPattern="[\w]+",
+    MinLength="16",
+    MaxLength="32",
 ))
 secret_access_key_param = template.add_parameter(Parameter(
     "AmazonSecretAccessKey",
-    Description="Your Amazon AWS secret access key.",
+    Description="Your Amazon AWS secret access key (mandatory)",
     Type="String",
+    NoEcho=True,
+    MinLength="1",
 ))
 keyname_param = template.add_parameter(Parameter(
     "EC2KeyPair",
     Description="Name of an existing EC2 KeyPair to enable SSH "
-                "access to the instance",
+                "access to the instance (mandatory)",
     Type="String",
```

Comment:
There's a special type for key pair name:
 * `AWS::EC2::KeyPair::KeyName`: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/parameters-section-structure.html
 * https://blogs.aws.amazon.com/application-management/post/Tx3DV2UYG9SC38G/Using-the-New-CloudFormation-Parameter-Types

Worth trying that?